### PR TITLE
set sound pin

### PR DESCRIPTION
**Pull request by pelikhan**
Add a new block, ``music.setSoundPin`` to control which pin the sound should generated to.

- [x] on v1, it has the same backward compat effect of ``analogSetPitchPin``
- [x] on v2, it enables generating sound to a pin via the mixer
- [x] on v2, analogSetPitchPin is deprecated and has the effect of bypassing the mixer (backward compat square waves)
image

This way: existing packages using analogSetPitchPin will get square waves; packages who want to leverage the mixer on a pin, can use this api.

### CODE REVIEW

File: libs/core/pins.cpp [hunk 1]
```diff
@@ -332,6 +332,7 @@ namespace pins {
     //% help=pins/analog-set-pitch-pin weight=3 advanced=true
     //% name.fieldEditor="gridpicker" name.fieldOptions.columns=4
     //% name.fieldOptions.tooltips="false" name.fieldOptions.width="250"
+    //% deprecated
```
Comment (jwunderl) [inline libs/core/pins.cpp:335]: Do we need to worry about the blocks being removed mid year? I'm guessing it's probably fine / unlikely that teachers were using these

**Review by jwunderl — APPROVED** [review]

File: libs/core/pins.cpp (hunk 1 above)
Comment (pelikhan) [reply libs/core/pins.cpp:335]: it's an advanced API that most beginners will never find (advanced section in pins under advanced) so hiding to really discourage using it. We'll see,

Comment (pelikhan) [reply libs/core/pins.cpp:335]: on the other hadn, does not hurt to keep it around eighter.

**Comment by abchatra** [comment]
Sorry late here. I think this block belongs in Pins & not Music. Student wouldn't understand what Pin means in a Music category.

image

**Pull request merged**